homeassistant/components/synology_dsm/__init__.py

```diff
@@ -77,13 +78,24 @@ async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry):
     hass.data.setdefault(DOMAIN, {})
     hass.data[DOMAIN][entry.unique_id] = api
 
+    # For SSDP compat
+    if not entry.data.get(CONF_MAC):
+        hass.async_create_task(_add_macs_to_entry(hass, entry, api))
+
     hass.async_create_task(
         hass.config_entries.async_forward_entry_setup(entry, "sensor")
     )
 
     return True
 
 
+async def _add_macs_to_entry(hass, entry, api):
+    network = api.dsm.network
```

Comment:
Having this I/O log once when updating the entry  :
```shell
2020-05-03 00:34:57 WARNING (MainThread) [homeassistant.util.async_] Detected I/O inside the event loop. This is causing stability issues. Please report issue for synology_dsm doing I/O at homeassistant/components/synology_dsm/__init__.py, line 93: network = api.dsm.network
```

How can I fix that ? `dsm.network` is a property.

Reply:
Wrap it in a function to make a callback

```
network = hass.async_add_executor_job(get_network)

def get_network(api):
  """Write network in a callable."""
  return api.dsm.network
```

Reply:
That was actually what I was thinking while doing `_add_macs_to_entry`

Reply:
I/O in properties always get us.

Alternative:

```python
network = await hass.async_add_executor_job(getattr, api.dsm, "network")
```

Reply:
Thanks, this resolve the issue